src/main/java/com/microsoft/cse/helium/app/controllers/ActorsController.java

```diff
@@ -3,116 +3,138 @@
           .defaultIfEmpty(ResponseEntity.notFound().build());
     } else {
       logger.error("Invalid actorId parameter " + actorId);
-      return Mono.justOrEmpty(ResponseEntity.badRequest().contentType(MediaType.TEXT_PLAIN)
-          .build());
+      return Mono.justOrEmpty(
+          ResponseEntity.badRequest().contentType(MediaType.TEXT_PLAIN).build());
     }
-
   }
 
-  /**
-   * getAllActors.
-   */
-  @RequestMapping(value = "", method = RequestMethod.GET,
+  /** getAllActors. */
+  @RequestMapping(
+      value = "",
+      method = RequestMethod.GET,
       produces = MediaType.APPLICATION_JSON_VALUE)
   @ApiOperation(value = "Get all actors", notes = "Retrieve and return all actors")
   @ApiResponses(value = {@ApiResponse(code = 200, message = "List of actor objects")})
-  public Flux<Actor> getAllActors(
-      @ApiParam(value = "(query) (optional) The term used to search Actor name", required = false)
-      @RequestParam final Optional<String> q,
-      @RequestParam("q") final Optional<String> query,
-      @ApiParam(value = "0 based page index", defaultValue = "0")
-      @RequestParam Optional<Integer> pageNumber,
-      @ApiParam(value = "page size (1000 max)", defaultValue = "100")
-      @RequestParam Optional<Integer> pageSize,
-      ServerHttpResponse response) {
-    Integer tmpPageNumber = 0;
-    Integer tmpPageSize = Constants.DEFAULT_PAGE_SIZE;
+  public Object getAllActors(
+      @ApiParam(value = "(query) (optional) The term used to search Actor name") @RequestParam("q")
+          final Optional<String> query,
+      @ApiParam(value = "1 based page index", defaultValue = "1") @RequestParam
+          Optional<String> pageNumber,
+      @ApiParam(value = "page size (1000 max)", defaultValue = "100") @RequestParam
+          Optional<String> pageSize) {
+
+    String q = null;
+
+    if (query.isPresent() && !StringUtils.isEmpty(query.get())) {
+      if (query.get() != null && !query.get().isEmpty()) {
+        if (validator.isValidSearchQuery(query.get())) {
+          q = query.get().trim().toLowerCase().replace("'", "''");
+        } else {
+          logger.error("Invalid q(search) parameter");
+          MultiValueMap<String, String> headers = new HttpHeaders();
+          headers.put("content-type", Arrays.asList("text/plain"));
+          return new ResponseEntity<>(
+              "Invalid q(search) parameter", headers, HttpStatus.BAD_REQUEST);
+        }
+      }
+    }
 
+    Integer pageNo = 0;
     if (pageNumber.isPresent() && !StringUtils.isEmpty(pageNumber.get())) {
-      if (pageNumber.get() >= 1 && pageNumber.get() <= Constants.MAX_PAGE_COUNT) {
-        tmpPageNumber = pageNumber.get();
+      if (!validator.isValidPageNumber(pageNumber.get())) {
+        logger.error("Invalid pageNumber parameter");
+        MultiValueMap<String, String> headers = new HttpHeaders();
+        headers.put("content-type", Arrays.asList("text/plain"));
```

Comment:
Shouldn't this be Content-type (note upper case)?

Reply:
Isn't there a MediaType.TEXT_PLAIN constant to use?  If so, please change it to use that.